server/src/sim/autofit.py

```diff
@@ -10,83 +10,110 @@ def autofit(D, timelimit=None, maxiters=500, simstartyear=2000, simendyear=2015,
         
-        # Prevalence data
-        prev = [struct() for p in range(D.G.npops)]
-        for p in range(D.G.npops): 
-            prev[p].data = struct()
-            prev[p].model = struct()
-            prev[p].data.x, prev[p].data.y = extractdata(D.G.datayears, D.data.key.hivprev[0][p]) # The first 0 is for "best"
-            prev[p].model.x = S.tvec
-            prev[p].model.y = S.people[1:,p,:].sum(axis=0) / S.people[:,p,:].sum(axis=0) # This is prevalence
+
+        # Pull out Prevalence data
+        prev = [dict() for p in range(D['G']['npops'])]
+        for p in xrange(D['G']['npops']): 
+            prev[p]['data'] = dict()
+            prev[p]['model'] = dict()
+            prev[p]['data']['x'], prev[p]['data']['y'] = extractdata(D['G']['datayears'], D['data']['key']['hivprev'][0][p]) # The first 0 is for "best"
+            prev[p]['model']['x'] = S['tvec']
+            prev[p]['model']['y'] = S['people'][1:,p,:].sum(axis=0) / S['people'][:,p,:].sum(axis=0) # This is prevalence
+                                                        
+        #indicators = struct()
+#        indicatorkeys = ['death', 'newtreat', 'numtest','numinfect', 'dx']
+#        #datakeys = ['death', 'newtreat', 'numtest','numinfect', 'dx']
+#        for key in indicatorkeys:
+#            indicators[key] = dict()
+        [death, newtreat, numtest, numinfect, dx] = [[dict()], [dict()], [dict()], [dict()], [dict()]]        
+
         
+        # Pull out other indicators data
         mismatch = 0
         allmismatches = []
-        for base in [dx, prev]:
+
+        for base in [death, newtreat, numtest, numinfect, dx]:
+            base[0]['data'] = dict()
+            base[0]['model'] = dict()
+            base[0]['model']['x'] = S['tvec']
+            if base == death:
+                base[0]['data']['x'], base[0]['data']['y'] = extractdata(D['G']['datayears'], D['data']['opt']['death'][0])
+                base[0]['model']['y'] = S['death'].sum(axis=0)
+            elif base == newtreat:
+                base[0]['data']['x'], base[0]['data']['y'] = extractdata(D['G']['datayears'], D['data']['opt']['newtreat'][0])
+                base[0]['model']['y'] = S['newtx1'].sum(axis=0) + S['newtx2'].sum(axis=0)
+            elif base == numtest:
+                base[0]['data']['x'], base[0]['data']['y'] = extractdata(D['G']['datayears'], D['data']['opt']['numtest'][0])
+                base[0]['model']['y'] = D['M']['hivtest'].sum(axis=0)*S['people'].sum(axis=0).sum(axis=0) #testing rate x population
+            elif base == numinfect:
+                base[0]['data']['x'], base[0]['data']['y'] = extractdata(D['G']['datayears'], D['data']['opt']['numinfect'][0])
+                base[0]['model']['y'] = S['inci'].sum(axis=0)
+            elif base == dx:
+                base[0]['data']['x'], base[0]['data']['y'] = extractdata(D['G']['datayears'], D['data']['opt']['numdiag'][0])
+                base[0]['model']['y'] = S['dx'].sum(axis=0)
```

Comment:
Note: D['data']['opt'][indicator] are actually lists of length 1 and D['data']['opt'][indicator][0] does read the entire list.